src/Core/Authoring/PhysicalPackageFile.cs

```diff
@@ -89,9 +91,171 @@ public IEnumerable<FrameworkName> SupportedFrameworks
 
         public Stream GetStream()
         {
-            return _streamFactory != null ? _streamFactory() : File.OpenRead(SourcePath);
+            try
+            {
+                return _streamFactory != null ? _streamFactory() : File.OpenRead(SourcePath);
+            }
+            catch (PathTooLongException)
+            {
+                return Win32File.OpenRead(SourcePath);

```

Comment:
You want to make sure this runs only on windows and only on supported versions of windows (not sure if that is an applicable problem or not, need to check)